lib/http.js

```diff
@@ -64,7 +65,8 @@ HttpClient.prototype.buildRequest = function(rurl, data, exheaders, exoptions) {
     uri: curl,
     method: method,
     headers: headers,
-    followAllRedirects: true
+    followAllRedirects: true,
+    timeout: (exoptions && exoptions.timeout) || DEFAULT_TIMEOUT
```

Comment:
should check for undefined and timeout of 0 would be falsy and pick DEFAULT_TIMEOUT
Better to just add below this, if(exoptions && exoptions.timeout!=undefined){ options.timeout=exoptions.timeout}